test/client/fixtures/sandbox/code-instrumentation/getters-test.js

```diff
@@ -525,3 +525,56 @@ test('we should clean up html and remove extra namespaces from svg (GH-1083)', f
 
     strictEqual(getProperty(div, 'innerHTML'), nativeDiv.innerHTML);
 });
+
+test('we should not process element\'s properties if they do not exist (GH-1164)', function () {
+    var div             = document.createElement('div');
+    var svg             = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    var html            = '<a href="/path">link</a>';
+    var processedHtml   = processHtml(html);
+    var style           = 'body {}';
+    var processedStyle  = styleProcessor.process(style, null, true);
+    var textProperties  = ['innerText', 'text', 'textContent'];
+
+    div.appendChild(svg);
+
+    var svgHasInnerHTML = svg.innerHTML !== void 0;
+
+    if (svgHasInnerHTML)
+        strictEqual(getProperty(svg, 'innerHTML'), '');
+    else
+        strictEqual(getProperty(svg, 'innerHTML'), void 0);
+
+    setProperty(svg, 'innerHTML', html);
+
+    if (svgHasInnerHTML)
+        strictEqual(svg.innerHTML, processedHtml);
+    else
+        strictEqual(svg.innerHTML, html);
+
+    svg.innerHTML = '';
+
+    var svgHasOuterHTML = svg.outerHTML !== void 0;
+
+    if (svgHasOuterHTML)
+        strictEqual(getProperty(svg, 'outerHTML'), svg.outerHTML);
+    else
+        strictEqual(getProperty(svg, 'outerHTML'), void 0);
+
+    setProperty(svg, 'outerHTML', html);
+
+    if (svgHasOuterHTML)
+        strictEqual(div.innerHTML, processedHtml);
+    else
+        strictEqual(svg.outerHTML, html);
+
+    textProperties.forEach(function (property) {
+        var svgHasProperty = svg[property] !== void 0;
+
+        svg[property] = processedStyle;
```

Comment:
As for `innerHTML` and `outerHTML`:
Change to `setProperty(svg, property, style)`